### PR TITLE
Added a prison on an island

### PR DESCRIPTION
**Pull request by Night-Pryanik**
#### Summary
```SUMMARY: Content "Added a prison on an island."```

#### Purpose of change
[This](https://www.reddit.com/r/cataclysmdda/comments/br9mgo/cast_away_scenario/) Reddit post encouraged me to create an island-based prison. This location could be used as a starting location for Castaway scenario, or a more challenging version of Escape from Prison scenario.

#### Describe the solution
- [x] create an island;
- [x] create a prison;
- [x] fill prison with monsters and items;
- [x] make it a starting location for scenarios.

#### Describe alternatives you've considered
None.

#### Additional context

There is no special or awesome loot, or quests, or NPCs here, at least for now.

Edit: these pics are here only for historic reasons, the more recent pics are in the comments below.

изображение
изображение

### CODE REVIEW

**Comment by Night-Pryanik** [comment]
Decided to make it a little bigger - 4x3 instead of a 3x3. Also make my own dock instead of built-in.

**Comment by Night-Pryanik** [comment]
Some concrete trails, rocks here and there, slightly adjusted dock, and a helipad.
изображение

**Comment by curstwist** [comment]
nice additions!

**Comment by Night-Pryanik** [comment]
Once again I enlarged the island (I have big plans for it):
изображение

Added fence wire and barbed wire - they bordered the high-security zone.

**Comment by Night-Pryanik** [comment]
Added part of a prison itself, also added some administrative building.
изображение

File: data/json/mapgen/prison_1.json [hunk 1]
```diff
@@ -259,89 +259,152 @@
...
+        "!": [ "t_dock" ],
+        "^": [ "t_concrete" ],
+        "$": [ "t_brick_wall" ],
+        "?": [ "t_door_locked" ],
+        "(": [ "t_pavement_y" ],
+        ")": [ "t_guardrail" ],
+        "1": [ "t_window_bars_alarm" ],
+        "q": [ "t_floor" ],
+        "A": [ "t_floor" ],
+        "`": [
```
Comment (ralreegorganon) [inline data/json/mapgen/prison_1.json]: For these lists, you can do `[ [ "t_water_dp", 20 ], "t_water_sh" ]` since #28542.

Comment (Night-Pryanik) [reply data/json/mapgen/prison_1.json]: Oh, shoot, I totally forgot about this. Thanks!

**Comment by Mecares** [comment]
Great, can't wait to encounter it in a playthrough. Does it have half dug escape tunnel with a spoon as digging tool at the end?

**Comment by paulenka-aleh** [comment]
Not until spoon is given digging & mining qualities in game! :)

**Comment by Night-Pryanik** [comment]
> Does it have half dug escape tunnel with a spoon as digging tool at the end?

We need to add Rita Hayworth portrait to the game first.

**Comment by Night-Pryanik** [comment]
Another update: added officers' house, walking area, shower area, mess hall and a storage, clothes' cleaning area, made infirmary larger, added slight variations to the simple "grass-dirt" landscape. 

изображение

**Comment by Night-Pryanik** [comment]
Close-up view of the prison area
изображение

The prison itself is mostly done, I just don't know what to place in watchtowers.

**Comment by Night-Pryanik** [comment]
Added zombie cops, regular zombies, zombie brutes and z-9s spawn. Anyone else should I add?

**Comment by Night-Pryanik** [comment]
I'm not sure if this location could be a good starting location before we mainline Boats mod. Otherwise the only way to escape this island would be swim.

**Comment by paulenka-aleh** [comment]
> I'm not sure if this location could be a good starting location before we mainline Boats mod. Otherwise the only way to escape this island would be swim.

Well, first of all you're choosing location, so you should know what you're doing, right?

Secondly it's really good if you're plaing with boats. Alternatively start itself could be put into boats mod...

And finally it gives one more reason to mainline boats thus increasing interest in this task among contributors

**Comment by Night-Pryanik** [comment]
> Alternatively start itself could be put into boats mod...

Ok, I'll go this way.

**Comment by ZhilkinSerg** [comment]
> 
> 
> > Alternatively start itself could be put into boats mod...
> 
> Ok, I'll go this way.

I think we will mainline boats mod soon.

**Comment by davidpwbrown** [comment]
Yep #30816 go ahead :) this looks cool.

**Comment by Night-Pryanik** [comment]
Another screenshot to check-up
изображение

**Comment by Night-Pryanik** [comment]
I feel the location is ready to review. The watchtowers, the roof and (possible) second floor could be added in consequent PRs.

File: data/json/scenarios/scenarios.json [hunk 2]
```diff
@@ -243,6 +243,17 @@
...
+  {
+    "type": "scenario",
+    "ident": "alcatraz",
+    "name": "Challenge-Island Prison Break",
+    "points": -6,
+    "description": "You were delivered to some high-security prison right before the Cataclysm.  You almost managed to escape the walls of a prison...  Too bad it's located on a remote island, and now you need to find out how to escape it too.",
+    "allowed_locs": [ "alcatraz" ],
+    "start_name": "Island prison",
+    "professions": [ "convict", "death_row_convict" ],
+    "flags": [ "LONE_START", "CHALLENGE", "OUTDOOR" ]
```
Comment (ZhilkinSerg) [inline data/json/scenarios/scenarios.json:255]: I am not aware whether `OUTDOOR` flag in scenarios was supported when this was committed, but currently there is no support. You can use `ALLOW_OUTSIDE` for `sloc_prison_alcatraz` though.

Comment (Night-Pryanik) [reply data/json/scenarios/scenarios.json:255]: Ok, noted.

**Pull request merged**